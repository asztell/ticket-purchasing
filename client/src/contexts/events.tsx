import { createContext, useEffect, useMemo, useState } from "react";

export type Event = {
  id: string;
  name: string;
  location: string;
  ISODate: string;
  price: number;
};

export type EventsContextType = {
  events: Event[];
  error: string | null;
};

export const EventsContext = createContext<EventsContextType>(
  {} as EventsContextType
);

export function EventsProvider({ children }: { children: React.ReactNode }) {
  const [events, setEvents] = useState<Event[]>([]);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    // const url = `${document.location.origin}/events`;
    const url = `${process.env.REACT_APP_API}/events`;
    console.log("url", url);
    fetch(url)
      .then((res) => {
        if (!res.ok) {
          throw res;
        }
        return res.json();
      })
      .then((result) => {
        console.log("events", result);
        setEvents(result);
      })
      .catch((error) => {
        setError(error);
      });
  }, [setEvents]);

  const value = useMemo(() => {
    return {
      events,
      error,
    };
  }, [events, error]);

  return (
    <EventsContext.Provider value={value}>{children}</EventsContext.Provider>
  );
}
