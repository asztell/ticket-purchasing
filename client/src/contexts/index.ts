export * from './events'
export * from './ticketPurchasing'
