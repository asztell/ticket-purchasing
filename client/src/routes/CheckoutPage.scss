.Checkout-Page {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  & > .Billing {
    flex-basis: calc(66.6% - 10px);
    @media (max-width: 1000px) {
      flex-basis: 100%;
    }
  }
  & > .Summary {
    flex-basis: calc(33.3% - 10px);
    @media (max-width: 1000px) {
      flex-basis: 100%;
    }
  }
}
