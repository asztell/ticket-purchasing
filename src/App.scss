.App-Content {
  //   width: 1000px;
  width: 80vw;
  //   border: 1px solid red;
  margin-left: auto;
  margin-right: auto;
}

.App-Header {
  /* background-color: #282c34; */
  background-color: rgb(2, 108, 223);
  height: 60px;
  /* min-height: 100vh; */
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  margin-bottom: 50px;
  /* color: white; */
}
