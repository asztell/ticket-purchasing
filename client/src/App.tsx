import type { ReactElement } from 'react'
import { Outlet } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Auth0Provider } from '@auth0/auth0-react'
import messages from './lang/en-US.json'
import { TicketPurchasingProvider, EventsProvider } from './contexts'
import { Header } from './components'
import './App.scss'

export function App(): ReactElement {
  const flattenedMessages = flattenMessages(messages)

  const auth0Domain = process.env.REACT_APP_AUTH0_DOMAIN as string
  const auth0ClientId = process.env.REACT_APP_AUTH0_CLIENT_ID as string
  const auth0RedirectUri = process.env.REACT_APP_AUTH0_REDIRECT_URI as string

  console.log('auth0Domain', auth0Domain)
  console.log('auth0ClientId', auth0ClientId)
  console.log('auth0RedirectUri', auth0RedirectUri)

  return (
    <IntlProvider locale={navigator.language} messages={flattenedMessages}>
      <EventsProvider>
        <TicketPurchasingProvider>
          <Auth0Provider
            domain={auth0Domain}
            clientId={auth0ClientId}
            authorizationParams={{
              // audience: process.env.REACT_APP_AUTH0_AUDIENCE,
              // scope: "openid profile email",
              redirect_uri: auth0RedirectUri
            }}
          >
            <Header />
            <div className='App'>
              <div className='App-Content'>
                <Outlet />
              </div>
            </div>
          </Auth0Provider>
        </TicketPurchasingProvider>
      </EventsProvider>
    </IntlProvider>
  )
}

export const flattenMessages = (
  nestedMessages: Record<string, any>,
  prefix = ''
): Record<string, string> => {
  if (nestedMessages === null) {
    return {}
  }
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key]
    const prefixedKey = prefix !== '' ? `${prefix}.${key}` : key

    if (typeof value === 'string') {
      Object.assign(messages, { [prefixedKey]: value })
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey))
    }

    return messages
  }, {})
}
