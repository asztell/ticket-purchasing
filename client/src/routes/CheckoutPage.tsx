import {
  Billing,
  Summary,
  LinkButton,
  NavigationContainer
} from '../components'
import classnames from 'classnames'
import './CheckoutPage.scss'

export function CheckoutPage(): JSX.Element {
  const checkoutToEventsClassName = classnames(
    'Checkout-To-Events',
    'Left-Link-Button'
  )
  const checkoutToHomePageClassName = classnames(
    'Checkout-To-Home-Page',
    'Right-Link-Button'
  )
  return (
    <>
      <div className='Checkout-Page'>
        <Billing className='Billing' />
        <Summary className='Summary' />
      </div>
      <NavigationContainer>
        <LinkButton
          to='/events'
          label='Events'
          className={checkoutToEventsClassName}
        />
        <LinkButton
          to='/'
          label='Home'
          className={checkoutToHomePageClassName}
        />
      </NavigationContainer>
    </>
  )
}
