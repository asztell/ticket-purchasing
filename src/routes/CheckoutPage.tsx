import { Billing, Summary, LinkButton } from "../components";
import "./CheckoutPage.scss";

export function CheckoutPage() {
  return (
    <>
      <div className="Checkout-Page">
        <Billing className="Billing" />
        <Summary className="Summary" />
      </div>
      <LinkButton to="/events" label="Back to Events" />
      <LinkButton to="/" label="Home Page" />
    </>
  );
}
