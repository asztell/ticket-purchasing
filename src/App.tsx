import { Outlet } from "react-router-dom";
import { IntlProvider } from "react-intl";
// import { flatten } from "flat";
import messages from "./lang/en-US.json";
import { TicketPurchasingProvider } from "./contexts/ticketPurchasing";
import { EventsProvider } from "./contexts/events";
import "./App.scss";

export function App() {
  const flattenedMessages = flattenMessages(messages);
  console.log(flattenedMessages);
  return (
    <div className="App">
      <IntlProvider locale={navigator.language} messages={flattenedMessages}>
        <header className="App-Header"></header>
        <EventsProvider>
          <TicketPurchasingProvider>
            <div className="App-Content">
              <Outlet />
            </div>
          </TicketPurchasingProvider>
        </EventsProvider>
      </IntlProvider>
    </div>
  );
}

export const flattenMessages = (
  nestedMessages: Record<string, any>,
  prefix = ""
) => {
  if (nestedMessages === null) {
    return {};
  }
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (typeof value === "string") {
      Object.assign(messages, { [prefixedKey]: value });
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey));
    }

    return messages;
  }, {});
};
