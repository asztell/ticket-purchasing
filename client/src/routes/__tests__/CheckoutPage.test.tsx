import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { IntlProvider } from "react-intl";
import { flattenMessages } from "../../App";
import { CheckoutPage } from "../CheckoutPage";
import { TicketPurchasingContext } from "../../contexts";

describe("<CheckoutPage />", () => {
  test("render CheckoutPage", () => {
    const { container } = render(
      <Router>
        <IntlProvider
          locale="en"
          messages={flattenMessages({
            // TODO: create a test helper version of the flattenMessages function
            // that adds Test to the beginning of each message
            Billing: {
              Delivery: {
                Header: {
                  Text: "Test Delivery",
                },
                "Sub-Header": {
                  Text: "Test Mobile - Free",
                },
                Description: {
                  Text: "Test Your phone's your ticket. Locate your tickets in your account - or in your app. When you go mobile, your tickets will not be emailed to you or available for print.",
                },
              },
              Payment: {
                Header: {
                  Text: "Test Payment",
                },
                "Payment-Section": {
                  "Name-On-Card": {
                    Label: "Test Name on Card",
                  },
                  "Card-Number": {
                    Label: "Test Card Number",
                  },
                  "Expiration-Date": {
                    Label: "Test Expiration Date",
                  },
                  "Security-Code": {
                    Label: "Test Security Code",
                  },
                },
              },
              "Ticket-Insurance": {
                Header: {
                  Text: "Test Ticket Insurance",
                },
                Description: {
                  Text: "Test Get reimbursed up to 100% with Event Ticket Insurance for only $28.00 per ticket ($56.00 total). *Offer not available after purchase is finalized.",
                },
              },
            },
          })}
        >
          <TicketPurchasingContext.Provider
            value={{
              selectedEvent: {
                name: "Test Event",
                location: "Test Location",
                ISODate: "2021-01-01",
                price: 100,
              },
              ticketsCounter: 1,
              cardInfo: {
                cardNumber: "1234123412341234",
                nameOnCard: "Test User",
                cardType: "Test Card",
                securityCode: "123",
                securityCodeValid: true,
                expirationDate: "01/01/2021",
                expirationDateValid: true,
              },
              termsOfUseChecked: true,
              updateSelectedEvent: () => {},
              updateTicketsCounter: () => {},
              updateCardInfo: () => {},
              updateTermsOfUseChecked: () => {},
            }}
          >
            <CheckoutPage />
          </TicketPurchasingContext.Provider>
        </IntlProvider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
