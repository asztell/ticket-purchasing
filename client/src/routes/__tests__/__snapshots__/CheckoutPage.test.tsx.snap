// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutPage /> render CheckoutPage 1`] = `
<div>
  <div
    class="Checkout-Page"
  >
    <div
      class="Billing"
    >
      <div
        class="Delivery"
      >
        <h2
          class="Section-Header"
        >
          Test Delivery
        </h2>
        <h3>
          Test Mobile - Free
        </h3>
        <p>
          Test Your phone's your ticket. Locate your tickets in your account - or in your app. When you go mobile, your tickets will not be emailed to you or available for print.
        </p>
      </div>
      <div
        class="Payment"
      >
        <h2
          class="Section-Header"
        >
          Test Payment
        </h2>
        <div
          class="Accepted-Cards"
          data-tid="accepted-cards"
        >
          <div
            aria-label="american-express"
            class="Amex"
            role="img"
          />
          <div
            aria-label="visa"
            class="Visa"
            role="img"
          />
          <div
            aria-label="master-card"
            class="Master-Card"
            role="img"
          />
          <div
            aria-label="discover"
            class="Discover"
            role="img"
          />
          <div
            aria-label="diners-club"
            class="Diners-Club"
            role="img"
          />
        </div>
        <div
          class="Payment-Section"
        >
          <label
            class="Payment-Form-Label"
            for="name-on-card"
          >
            Test Name on Card
          </label>
          <input
            class="nameOnCard"
            type="text"
            value="Test User"
          />
        </div>
        <div
          class="Payment-Section"
        >
          <label
            class="Payment-Form-Label"
            for="card-number"
          >
            Test Card Number
          </label>
          <input
            class="cardNumber"
            type="text"
            value="1234123412341234"
          />
          <div
            class="Payment-Validation-Error"
          >
            Test Card
          </div>
        </div>
        <div
          class="Payment-Section Expiration-Date-And-Security-Code"
        >
          <div
            class="Expiration-Date"
          >
            <label
              class="Payment-Form-Label"
            >
              Test Expiration Date
            </label>
            <input
              class="expirationDate"
              maxlength="8"
              type="month"
              value="01/01/2021"
            />
            <div />
          </div>
          <div
            class="Security-Code"
          >
            <label
              class="Payment-Form-Label"
            >
              Test Security Code
            </label>
            <input
              class="securityCode"
              maxlength="3"
              type="text"
              value="123"
            />
            <div />
          </div>
        </div>
        <div />
      </div>
      <div
        class="Ticket-Insurance"
      >
        <h2
          class="Section-Header"
        >
          Test Ticket Insurance
        </h2>
        <p>
          Test Get reimbursed up to 100% with Event Ticket Insurance for only $28.00 per ticket ($56.00 total). *Offer not available after purchase is finalized.
        </p>
      </div>
    </div>
    <div
      class="Summary"
    >
      <div
        class="Total"
      >
        <h2>
          Total
        </h2>
        <p>
          Event: 
          Test Event
        </p>
        <p>
          Tickets: 
          1
        </p>
        <p>
          Card Number: 
          1234123412341234
        </p>
        <p>
          Security Code: 
          123
        </p>
        <input
          checked=""
          name="termsOfUse"
          type="checkbox"
        />
        <label
          for="termsOfUse"
        >
          I have read and agree to the current 
          <a
            href="#"
          >
            Terms of Use
          </a>
        </label>
        <button
          class="Purchase-Tickets-Button"
        >
          Place Order
        </button>
      </div>
    </div>
  </div>
  <div
    class="Navigation-Container"
  >
    <div>
      <button
        class="Link-Button Checkout-To-Events Left-Link-Button"
      >
        Events
      </button>
    </div>
    <div>
      <button
        class="Link-Button Checkout-To-Home-Page Right-Link-Button"
      >
        Home
      </button>
    </div>
  </div>
</div>
`;
